src/expr.rs

```diff
@@ -1602,7 +1636,17 @@ pub fn rewrite_assign_rhs<S: Into<String>>(context: &RewriteContext,
     };
     // 1 = space between operator and rhs.
     let max_width = try_opt!(width.checked_sub(last_line_width + 1));
-    let rhs = ex.rewrite(&context, max_width, offset + last_line_width + 1);
+    let mut rhs = ex.rewrite(&context, max_width, offset + last_line_width + 1);
```

Comment:
It seems this rewrite is wasted, could we only do it if we don't do the block rewrite?